ddl/rollingback.go

```diff
@@ -285,6 +285,43 @@ func rollingbackDropIndex(t *meta.Meta, job *model.Job) (ver int64, err error) {
 	return ver, errCancelledDDLJob
 }
 
+func rollingbackDropIndexes(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	tblInfo, indexInfos, ifExists, err := checkDropIndexes(t, job)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	indexInfo := indexInfos[0]
+	originalState := indexInfo.State
+	switch indexInfo.State {
+	case model.StateWriteOnly, model.StateDeleteOnly, model.StateDeleteReorganization, model.StateNone:
+		// We can not rollback now, so just continue to drop index.
+		// Normally won't fetch here, because there is check when cancel ddl jobs. see function: isJobRollbackable.
+		job.State = model.JobStateRunning
```

Comment:
I think we need to put this operation to `IsJobRollbackable` in `util/admin/admin.go`.

And please add tests to `TestCancelJob` in `ddl_worker_test.go`.

Reply:
@zimulala I added some test cases to `TestCancelJobs` and `TestIsJobRollbackable`

Reply:
@ou-bing 
Could we add some tests to `TestCancelJob` in `ddl_worker_test.go`.

Reply:
@zimulala I added three cases in `TestCancelJob `.